File: app/src/main/java/com/constraint/vagabond/main/search/MainContract.java
package com.constraint.vagabond.main.search;

import com.constraint.vagabond.data.entities.RecreationalAreaList;

public interface MainContract {

  interface presenter {

    void onDestroy();

    void onSearch(String query, String apiKey);
  }

  interface MainView {

    void showProgress();

    void hideProgress();

    void setDataToRecyclerView(RecreationalAreaList recAreasList);

    void onResponseFailure();
  }

//  interface GetRecAreasInteractor {
//
//    void getRecAreasList(OnFinishedListener onFinishedListener, String query);
//
//    interface OnFinishedListener {
//
//      void onFinished(RecreationalAreaList recAreasList);
//
//      void onFailure(Throwable t);
//    }
//  }
}

File: app/src/main/java/com/sierra/vagabond/main/search/MainContract.kt
package com.sierra.vagabond.main.search

import com.sierra.vagabond.data.entities.RecreationalAreaList

interface MainContract {

    interface Presenter {

        fun onDestroy()

        fun onSearch(query: String, apiKey: String)
    }

    interface MainView {

        fun showProgress()

        fun hideProgress()

        fun setDataToRecyclerView(recAreasList: RecreationalAreaList)

        fun onResponseFailure()
    }
}
